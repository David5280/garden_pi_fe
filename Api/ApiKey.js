export const apiKey = 'c3b27e2f72d0496c26da9270019b8ea8'