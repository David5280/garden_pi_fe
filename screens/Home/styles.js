import { StyleSheet, Dimensions } from 'react-native';

export default StyleSheet.create({
  screenContainer: {
    alignItems: 'center',
    height: Dimensions.get('window').height,
    paddingTop: 50,
    width: Dimensions.get('window').width,
  },
  infoContainer: {
    alignItems: 'center',
    backgroundColor: 'rgba(115, 166, 134, .9)',
    padding: 10,
    margin: 5,
    width: Dimensions.get('window').width * .9,
    borderColor: '#A14550',
    borderRadius: 30,
    borderWidth: 2,
  },
  header: {
    paddingTop: 200
  },
  carouselScroll: {
    height: 200
  },
  forecastContainer: {
    flexDirection: 'column',
    backgroundColor: '#d5fdd5',
    borderColor: '#A14550',
    borderRadius: 30,
    borderWidth: 2,
  },
  label: {
    fontSize: 13,
    marginTop: 5,
  },
  bold: {
    fontSize: 15,
    fontWeight: 'bold'
  },
  graphLabel: {
    height: 25,
    marginTop: 25,
    padding: 2,
    alignItems: 'center',
    width: Dimensions.get('window').width * .55,
    marginLeft: 88
  },
  currentWidgets: {
    flexDirection: 'row'
  },
  currentSoilStatContainer: {
    alignItems: 'center',
    alignItems: 'center',
    backgroundColor: '#d5fdd5',
    borderColor: '#A14550',
    borderRadius: 30,
    borderWidth: 2,
    justifyContent: 'space-between',
    margin: 3,
    padding: 10,
    width: 170
  }
})