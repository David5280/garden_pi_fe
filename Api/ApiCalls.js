import { apiKey } from './ApiKey'

export const fetchWeather = () => {
  return fetch(`https://api.darksky.net/forecast/${apiKey}/39.751048,-104.996659`)
  .then(res => res.json())
  .then(res => res)
  .catch(error => error.message)
}