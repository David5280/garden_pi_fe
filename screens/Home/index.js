import React, { Component } from 'react';
import { View, Dimensions, TouchableOpacity, Text, Image } from 'react-native';
import styles from './styles';

export default class Home extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.splashHeader}>Welcome to GardenPi</Text>
        <Image 
            source={require('../../assets/images/sprout.png')}
            style={styles.plantIcon} />
        <TouchableOpacity style={styles.splashEnterBtn}>
          <Text>
            Enter your Garden...
          </Text>
        </TouchableOpacity>
      </View>
    )
  }
}