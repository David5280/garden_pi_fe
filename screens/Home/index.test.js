import React from 'react';
import Home from './index';
import 'react-native';
import renderer from 'react-test-renderer';
import * as constants from '../../Api/ApiCalls'

jest.mock('react-native-gesture-handler', () => {});
jest.mock("expo-camera", () => {
  return {}
});
jest.mock('react-native-chart-kit', () => {
  return {}
})


// test('renders correctly', () => {
//   const tree = renderer.create(<Home />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

test('onWaterPress should post to triggerWaterJobs', async () => {
  constants.triggerWaterJob = jest.fn()
  constants.triggerWaterJob()
  expect(constants.triggerWaterJob).toHaveBeenCalled()
})