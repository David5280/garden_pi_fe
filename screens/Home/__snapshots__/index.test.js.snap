// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "height": 1334,
        "paddingTop": 50,
        "width": 750,
      }
    }
  >
    <Image
      source=""
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(115, 166, 134, .7)",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "justifyContent": "center",
            "margin": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "width": 712.5,
          },
          Object {
            "height": 306.82,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "auto",
            "width": 600,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source=""
            style={
              Array [
                Object {
                  "height": 35,
                  "width": 35,
                },
                Object {
                  "marginRight": 25,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "lemondrop",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "textShadowRadius": 5,
                },
                Object {
                  "color": "#E64C3C",
                  "fontSize": 35,
                },
              ]
            }
          >
            G
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "lemondrop",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "textShadowRadius": 5,
                },
                Object {
                  "color": "#FF9933",
                  "fontSize": 35,
                },
              ]
            }
          >
            A
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "lemondrop",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "textShadowRadius": 5,
                },
                Object {
                  "color": "#F1C431",
                  "fontSize": 35,
                },
              ]
            }
          >
            R
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "lemondrop",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "textShadowRadius": 5,
                },
                Object {
                  "color": "#187F03",
                  "fontSize": 35,
                },
              ]
            }
          >
            D
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "lemondrop",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "textShadowRadius": 5,
                },
                Object {
                  "color": "#3998DB",
                  "fontSize": 35,
                },
              ]
            }
          >
            E
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "lemondrop",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "textShadowRadius": 5,
                },
                Object {
                  "color": "#4b0082",
                  "fontSize": 35,
                },
              ]
            }
          >
            N
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "lemondrop",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "textShadowRadius": 5,
                },
                Object {
                  "color": "#7f00ff",
                  "fontSize": 35,
                },
              ]
            }
          >
            Pi
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source=""
            style={
              Array [
                Object {
                  "height": 35,
                  "width": 35,
                },
                Object {
                  "marginLeft": 25,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 133.4,
            "justifyContent": "center",
            "marginBottom": "auto",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "auto",
            "padding": 2,
            "width": 487.5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          No 
          Weather
           Data Available.
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(115, 166, 134, .7)",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "justifyContent": "center",
            "margin": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "width": 712.5,
          },
          Object {
            "height": 413.54,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d5fdd5",
              "borderColor": "#A14550",
              "borderRadius": 30,
              "borderWidth": 2,
              "height": 133.4,
              "justifyContent": "center",
              "marginBottom": "auto",
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": "auto",
              "padding": 2,
              "width": 487.5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            No 
            Soil Moisture
             Data Available.
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(115, 166, 134, .7)",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "justifyContent": "center",
            "margin": 5,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 1,
            "width": 712.5,
          },
          Object {
            "height": 306.82,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d5fdd5",
            "borderColor": "#A14550",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 133.4,
            "justifyContent": "center",
            "marginBottom": "auto",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": "auto",
            "padding": 2,
            "width": 487.5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          No 
          Soil
           Data Available.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 675,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAF0E6",
            "borderColor": "#A14550",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 75,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 5,
            "opacity": 1,
            "overflow": "hidden",
            "width": 300,
          }
        }
      >
        <Text>
          More Data
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAF0E6",
            "borderColor": "#A14550",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 75,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 5,
            "opacity": 1,
            "overflow": "hidden",
            "width": 300,
          }
        }
      >
        <Text>
          Water Garden
        </Text>
      </View>
    </View>
  </View>
</View>
`;
